<?xml version="1.0"?>
<launch>



  <node pkg="amcl" type="amcl" name="amcl">
      <param name="odom_model_type" value="omni"/>
      <param name="odom_alpha5" value="0.1"/>
      
<!--Lowering transform_tolerance value may lead to Expolration Error. Adjust this value according to the processor capacity -->
      <param name="transform_tolerance" value="0.5" /> 
      
      <param name="gui_publish_rate" value="10.0"/> 
      <param name="laser_max_beams" value="30"/>
      <param name="min_particles" value="500"/>
      <param name="max_particles" value="2000"/>
      <param name="kld_err" value="0.05"/>
      <param name="kld_z" value="0.99"/>
      <param name="odom_alpha1" value="0.2"/>
      <param name="odom_alpha2" value="0.2"/> 
      <!-- translation std dev, m -->
      <param name="odom_alpha3" value="0.8"/>
      <param name="odom_alpha4" value="0.2"/>
      <param name="laser_z_hit" value="0.5"/>
      <param name="laser_z_short" value="0.05"/>
      <param name="laser_z_max" value="0.05"/>
      <param name="laser_z_rand" value="0.5"/>
      <param name="laser_sigma_hit" value="0.2"/>
      <param name="laser_lambda_short" value="0.1"/>
      <param name="laser_lambda_short" value="0.1"/>
      <param name="laser_model_type" value="likelihood_field"/>
      <param name="laser_likelihood_max_dist" value="2.0"/>
      <param name="update_min_d" value="0.1"/>
      <param name="update_min_a" value="0.2"/>
      <param name="odom_frame_id" value="odom"/>
      <param name="resample_interval" value="1"/>
      <param name="recovery_alpha_slow" value="0.0"/>
      <param name="recovery_alpha_fast" value="0.0"/> 
  </node>

  <arg name="base_global_planner" default="navfn/NavfnROS"/>
  <arg name="base_local_planner" default="dwa_local_planner/DWAPlannerROS"/>
  
  <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen">

    <param name="base_global_planner" value="$(arg base_global_planner)"/>
    <param name="base_local_planner" value="$(arg base_local_planner)"/>  
    
    <rosparam file="$(find turtlebot3_2dnav)/map/planner.yaml" command="load"/>

    <rosparam file="$(find turtlebot3_2dnav)/map/costmap_common_params.yaml" command="load" ns="global_costmap" />
    <rosparam file="$(find turtlebot3_2dnav)/map/costmap_common_params.yaml" command="load" ns="local_costmap" />
    
    <rosparam file="$(find turtlebot3_2dnav)/map/local_costmap_params.yaml" command="load" ns="local_costmap" />
    <rosparam file="$(find turtlebot3_2dnav)/map/global_costmap_params.yaml" command="load" ns="global_costmap" /> 

  </node> 
  
</launch>